tests/test_sync_security.py

```
@@ -0,0 +1,394 @@
+    request = urllib.request.Request(
+        f"{remote}/sync/push",
+        data=body,
+        headers={
+            "Authorization": "Bearer relay-key",
+            "Content-Type": "application/json",
+            "X-Mnemosyne-Body-MAC": hmac.new(
+                b"relay-key", body, hashlib.sha256
+            ).hexdigest(),
+        },
+        method="POST",
+    )
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)